src/tdycoremesh.c

```diff
@@ -0,0 +1,1230 @@
+#include <petsc.h>
+#include <private/tdycoreimpl.h>
+#include <private/tdymeshimpl.h>
+#include <private/tdymemoryimpl.h>
+#include <private/tdyutils.h>
+
+/* ---------------------------------------------------------------- */
+
+PetscBool IsClosureWithinBounds(PetscInt closure, PetscInt start,
+                                PetscInt end) {
+  return (closure >= start) && (closure < end);
+}
+
+/* -------------------------------------------------------------------------- */
+PetscErrorCode AllocateMemoryForASubcell(
+  TDy_subcell    *subcell,
+  TDySubcellType subcell_type) {
+
+  PetscFunctionBegin;
+
+  PetscInt num_nu_vectors;
+  PetscInt num_vertices;
+
+  switch (subcell_type) {
+  case SUBCELL_QUAD_TYPE:
+    num_nu_vectors = 2;
+    num_vertices   = 4;
+    break;
+  case SUBCELL_HEX_TYPE:
+    num_nu_vectors = 3;
+    num_vertices   = 8;
+    break;
+  }
+
+  subcell->type           = subcell_type;
+  subcell->num_nu_vectors = num_nu_vectors;
+  subcell->num_vertices   = num_vertices;
+
+  subcell->nu_vector                       = (TDy_vector *) malloc(
+        num_nu_vectors * sizeof(TDy_vector    ));
+  subcell->variable_continuity_coordinates = (TDy_coordinate *) malloc(
+        num_nu_vectors * sizeof(TDy_coordinate));
+  subcell->vertices_cordinates             = (TDy_coordinate *) malloc(
+        num_vertices   * sizeof(TDy_coordinate));
+
+  PetscFunctionReturn(0);
+}
+
+/* -------------------------------------------------------------------------- */
+PetscErrorCode AllocateMemoryForACell(
+  TDy_cell       *cell,
+  PetscInt       num_vertices,
+  PetscInt       num_edges,
+  PetscInt       num_neighbors,
+  TDySubcellType subcell_type) {
+
+  PetscFunctionBegin;
+
+  PetscErrorCode ierr;
+  PetscInt       isubcell;
+  PetscInt       num_subcells;
```

Comment:
```suggestion
  PetscInt       num_subcells=0;
```

Reply:
Similarly, I added an error message under the `default` case within the `switch(subcell_type)` https://github.com/TDycores-Project/TDycore/pull/5/commits/e7441c7e94d399ab7f76cee3617631efa9a1ae3a#diff-fcf52e8a4105ab2e05e6a4f90c5dfbd9R87